pkgs/_macro_client/lib/macro_client.dart

```diff
@@ -2,12 +2,36 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:convert';
+import 'dart:io';
+
 import 'package:macro/macro.dart';
 import 'package:macro_service/macro_service.dart';
 
+/// Runs macros, connecting them to a macro host.
 class MacroClient {
+  final Iterable<Macro> macros;
+  final Socket socket;
```

Comment:
It isn't clear to me why these are instance variables, or why this class exists, as the state isn't used at all after the constructor.

The lifecycle of the `socket` is also confusing to me here, who is in charge of closing it? When and how does it get closed?